server/src/test/scala/org/http4s/server/ServerSpec.scala

```diff
@@ -0,0 +1,45 @@
+package org.http4s
+package server
+
+import java.net.{InetSocketAddress, URL}
+import fs2.Task
+import org.http4s.dsl._
+import org.specs2.specification.AfterAll
+import scala.concurrent.ExecutionContext
+import scala.io.Source
+
+trait ServerContext extends AfterAll {
+  def builder: ServerBuilder
+
+  lazy val server = builder.bindAny()
+    .withExecutionContext(ExecutionContext.global)
+    .mountService(HttpService {
+      case GET -> Root / "thread" / "routing" =>
+        val thread = Thread.currentThread.getName
+        Ok(thread)
+      
+      case GET -> Root / "thread" / "effect" =>
+        Task.delay(Thread.currentThread.getName).flatMap(Ok(_))
+    })
+    .start
+    .unsafeRun
```

Comment:
Not sure I like it this being in a lazy val, but it proves what we'd like for the tests, as well as abstracting concerns from the test sites.